### ISSUE TITLE
Implement a single trainer

### ISSUE THREAD

**Issue by RistoAle97** [RistoAle97/ContinualNAT-NMT#9]
- The training is performed by using the script `train.py` right now and the user must manually change the model, dataset collators and callbacks. A trainer class should be built in order to have a more general and automatized approach.
- There is still an issue during the validation loss logging (but this does not impact the model's performance) while using gradient accumulation, this should be solved.
- The progress bar used during training should show the actual number of stepping batches instead of the estimated ones, this implies that a custom progress bar must be built.
- It must be possible to not only train on a single direction (e.g.: en->es) but also on both ways given a pair of languages (e.g.: en<->es).
- If more than two languages are passed, then the user should have the choice on the translation directions.
